uwblueprint/planet-read
# Authenticate users for review/translation platforms

## Notion ticket link



[Authenticate users for review/translation platforms](https://www.notion.so/uwblueprintexecs/Authenticate-users-for-review-translation-platforms-58cbbacb73374ba7b84c2a7231cb75b1)



## Implementation description

- Get `translatorId` and `reviewerId` in query when translation/review page is loaded, and check if it matches the id of the currently logged in user
- Show `NotFound` page if id does not match



## Steps to test

1. Log in as a user with both translation/review stories (e.g. Dwight D Eisenhower)
2. Click on view translation on of their translation stories and verify that the translation page loads
3. Enter the same URL manually and verify that it loads
4. Switch the URL from `/translation/` to `/review/` and verify that the page doesn't load
5. Try selecting a new story to translate and verify that the translation page loads
6. Repeat steps 2-5 but for the review platform



## What should reviewers focus on?

- `TranslationPage.tsx` and `ReviewPage.tsx`  
- Before the query is completed, only an empty div should be returned (this was done to prevent the 404 from showing right away)--is there something else that should be rendered instead?

## Checklist
